waku/v2/node/rest/store/handlers.nim

```diff
@@ -0,0 +1,251 @@
+    let msg = "Failed parsing remote peer info [" & e.msg & "]"
+    return err(msg)
+
+  return ok(storenode)
+
+# Creates a HistoryQuery from the given params
+proc createHistoryQuery(pubsubTopic: Option[string],
+                        contentTopics: Option[string],
+                        senderTime: Option[string],
+                        storeTime: Option[string],
+                        digest: Option[string],
+                        startTime: Option[string],
+                        endTime: Option[string],
+                        pageSize: Option[string],
+                        ascending: Option[string]):
+
+                  Result[HistoryQuery, string] =
+
+  # Parse pubsubTopic parameter
+  var parsedPubsubTopic = none(string)
+  if pubsubTopic.isSome():
+    let decodedPubsubTopic = decodeUrl(pubsubTopic.get())
+    if decodedPubsubTopic != "":
+      parsedPubsubTopic = some(decodedPubsubTopic)
+    debug "Setting pubsubTopic ", pubsub_topic=parsedPubsubTopic.get()
+
+  # Parse the content topics
+  var parsedContentTopics = newSeq[ContentTopic](0)
+  if contentTopics.isSome():
+    let ctList = decodeUrl(contentTopics.get())
+    if ctList != "":
+      for ct in ctList.split(','):
+        parsedContentTopics.add(ct)
+
+  # Parse cursor information
+  var parsedCursor = none(HistoryCursor)
+  let createCursor = parsedPubsubTopic.isSome() and
+                     senderTime.isValidTimestamp() and
+                     storeTime.isValidTimestamp() and
+                     digest.isSome() and digest.get() != ""
+  if createCursor:
+    parsedCursor = some(
+            HistoryCursor(
+                pubsubTopic: parsedPubsubTopic.get(),
+                senderTime: senderTime.get().toTimestamp(),
+                storeTime: storeTime.get().toTimestamp(),
+                digest: digest.get().toMessageDigest()
+                      ))
+
+  # Parse page size field
+  var parsedPagedSize = DefaultPageSize
+  if pageSize.isSome() and pageSize.get() != "":
+    try:
+      parsedPagedSize = uint64(parseInt(pageSize.get()))
+    except CatchableError as e:
+      return err("Problem parsing page size [" & e.msg & "]")
+
+  # Parse start time
+  var parsedStartTime = none(Timestamp)
+  if startTime.isSome() and startTime.get() != "":
+    try:
+      let time = parseInt(startTime.get())
+      if time > 0:
+        parsedStartTime = some(Timestamp(time))
+    except CatchableError as e:
+      return err("Problem parsing start time [" & e.msg & "]")
+
+  # Parse end time
+  var parsedEndTime = none(Timestamp)
+  if endTime.isSome() and endTime.get() != "":
+    try:
+      let time = parseInt(endTime.get())
+      if time > 0:
+        parsedEndTime = some(Timestamp(time))
+    except CatchableError as e:
+      return err("Problem parsing end time [" & e.msg & "]")
+
+  # Parse ascending field
+  var parsedAscending = true
+  if ascending.isSome() and ascending.get() != "":
+    parsedAscending = ascending.get() == "1"
```

Comment:
The `ascending` field must be `true` or `false` (default value is not present). We should avoid the `1/0 on/off yes/no true/false` support war.

Reply:
Completely agree but I couldn't manage to use an Option[bool] given how the nim-presto library parses the route-api code, which only accepts string types: 

> https://github.com/status-im/nim-presto/blob/2b440a443f3fc29197f267879e16bb8057ccc0ed/presto/route.nim#L322

For now I can only make it to handle "true" or "false" strings.